merlin/models/tf/models/ranking.py

```diff
@@ -152,3 +160,68 @@ def DCNModel(
 
 def YoutubeDNNRankingModel(schema: Schema) -> Model:
     raise NotImplementedError()
+
+
+def DeepFMModel(
+    schema: Schema,
+    embedding_dim: int,
+    deep_block: Optional[Block] = MLPBlock([64, 128]),
+    prediction_tasks: Optional[
+        Union[PredictionTask, List[PredictionTask], ParallelPredictionBlock]
+    ] = None,
+    **kwargs
+):
+    """DeepFM-model architecture.
+
+    Example Usage::
+        depp_fm = DeepFMModel(schema, embedding_dim=64, deep_block=MLPBlock([256, 64]))
+        depp_fm.compile(optimizer="adam")
+        depp_fm.fit(train_data, epochs=10)
+
+    References
+    ----------
+    [1] Huifeng, Guo, et al.
+        "DeepFM: A Factorization-Machine based Neural Network for CTR Prediction"
+        arXiv:1703.04247  (2017).
+
+    Parameters
+    ----------
+    schema : Schema
+        The `Schema` with the input features
+    embedding_dim : int
+        Dimension of the embeddings
+    deep_block : Block
+        The `Block` that learns high-ordeer feature interactions
+    prediction_tasks: optional
+        The prediction tasks to be used, by default this will be inferred from the Schema.
+
+    Returns
+    -------
+    Model
+
+    """
+    num_features = len(schema.column_names)
+    cardinalities = schema_utils.categorical_cardinalities(schema)
+
+    input_block = InputBlock(
+        schema, embedding_options=EmbeddingOptions(embedding_dim_default=embedding_dim), **kwargs
```

Comment:
I’ve seen this pattern in a few different places lately, not exactly sure if it’s the best way of doing this. This because we are only exposing one argument of `EmbeddingOptions`. Should we maybe add an extra argument (maybe like `embedding_options_kwargs`) to `DeepFMModel`?

Reply:
Yes make sense, the user could have more flexibility on how to set the embeddings. I've added `embedding_options_kwargs`.